capsules/extra/src/symmetric_encryption/aes.rs

```diff
@@ -134,21 +128,15 @@ impl<
                                             AesOperation::AES128Ctr(_)
                                             | AesOperation::AES128CBC(_)
                                             | AesOperation::AES128ECB(_) => {
-                                                if let Err(e) = AES128::set_key(self.aes, buf) {
-                                                    return Err(e);
-                                                }
+                                                AES128::set_key(self.aes, buf)?;
```

Comment:
This whole block can be simplified dramatically. I can take a pass, or we can avoid additional changes in this PR, and just focus on making clippy happy, cleaning up code style even more separately.